Fix TestFile.test_directory_listing

Currently this test is disabled (with `@pytest.mark.xfail`) because it fails.  If I remove the pytest mark and run it, I get:

```
$ tox -- tests/checker/test_file.py
...
=================================== FAILURES ===================================
_______________________ TestFile.test_directory_listing ________________________

self = <tests.checker.test_file.TestFile testMethod=test_directory_listing>

    def test_directory_listing (self):
        # unpack non-unicode filename which cannot be stored
        # in the SF subversion repository
        if os.name != 'posix' or sys.platform != 'linux2':
            return
        dirname = get_file("dir")
        if not os.path.isdir(dirname):
            unzip(dirname + ".zip", os.path.dirname(dirname))
>       self.file_test("dir")

tests/checker/test_file.py:94: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/checker/__init__.py:199: in file_test
    self.fail_unicode(msg)
tests/checker/__init__.py:205: in fail_unicode
    self.fail(msg)
E   AssertionError: tests/checker/data/dir
E   --- 
E   
E   +++ 
E   
E   @@ -3,8 +3,8 @@
E   
E    real url file:///home/mg/src/linkchecker/tests/checker/data/dir/
E    name tests/checker/data/dir
E    valid
E   -url %C3%AD%C2%BB%C2%AD%C2%AF%C2%BF.dat
E   -cache key file:///home/mg/src/linkchecker/tests/checker/data/dir/%C3%AD%C2%BB%C2%AD%C2%AF%C2%BF.dat
E   -real url file:///home/mg/src/linkchecker/tests/checker/data/dir/%C3%AD%C2%BB%C2%AD%C2%AF%C2%BF.dat
E   -name ??????????.dat
E   +url %ED%BB%AD%AF%BF.dat
E   +cache key file:///home/mg/src/linkchecker/tests/checker/data/dir/%ED%BB%AD%AF%BF.dat
E   +real url file:///home/mg/src/linkchecker/tests/checker/data/dir/%ED%BB%AD%AF%BF.dat
E   +name ?????.dat
E    valid
================ 1 failed, 14 passed, 2 skipped in 0.53 seconds ================
ERROR: InvocationError: '/home/mg/src/linkchecker/.tox/py27/bin/py.test tests/checker/test_file.py'
___________________________________ summary ____________________________________
ERROR:   py27: commands failed
```

--- a/tests/checker/test_file.py
+++ b/tests/checker/test_file.py
@@ -20,8 +20,6 @@ import os
 import sys
 import zipfile
 
-import pytest
-
 from tests import need_network, need_word, need_pdflib
 from . import LinkCheckTest, get_file
 
@@ -87,7 +85,6 @@ class TestFile(LinkCheckTest):
     def test_urllist(self):
         self.file_test("urllist.txt")
 
-    @pytest.mark.xfail
     def test_directory_listing(self):
         # unpack non-unicode filename which cannot be stored
         # in the SF subversion repository

